### DISCUSSION TITLE
zh/algorithms/hashring

### DISCUSSION THREAD

**Discussion by giscus** [selfboot/ai_gallery#15] [Q&A]
# zh/algorithms/hashring

友好的一致性 hash 环可视化演示，支持 hash 的各种操作，交互式演示方便理解节点添加、移除和键分配迁移过程。

https://gallery.selfboot.cn/zh/algorithms/hashring

**Comment by a-shy-coder** [comment]
请问引入虚拟节点后, 如果移除一个物理节点(红色), 该物理节点上分配的key, 是按顺时针找到下一个虚拟节点(?色), 然后分配给该虚拟节点对应的物理节点吗?(红色->?色)
 (为什么网站有虚拟节点的动画演示并不是这样?)

**Reply by 1967949440** [reply]
顺时针遇到的第一个虚拟节点
